### ISSUE TITLE
Save and download CSV file from API response.

### ISSUE THREAD

**Issue by kavitha24** [cypress-io/cypress#6456]
My API response contains .xls file . Like Postman (which provides save and download option and saves the response as file) does Cypress provides such way.

Desired Behaviour: 

I don't find a way to download the CSV file from API response.

**Comment by jennifer-shehane** [comment]
When opening an issue, please fill out the [issue template](https://github.com/cypress-io/cypress/blob/develop/.github/ISSUE_TEMPLATE.md) provided - it requires necessary information we need to help you, like desired behavior, a reproducible example and the version of Cypress you are using. 

This issue will be closed as per our [contributing guidelines](https://github.com/cypress-io/cypress/blob/develop/.github/CONTRIBUTING.md#does-the-issue-provide-all-the-information-from-our-issue-template). Please comment in this issue with the required information from our issue template and we will reopen the issue. 🙏

**Comment by kavitha24** [comment]
Scenario:  I am saving the API response which returns bytes as CSV file like below cy.writeFile('cypress/fixtures/metafile.csv',response.body,'utf8')
when I opened the CSV, it is not having proper values instead it shows data something like below.
Why it saves the values like these even I used the encoding option as utf8.

Q3VzdG9tZXIgTmFtZSxBZHZpc29yIE1wbiBJZCxCaWxsaW5nIE1wbiBJZCxQcmltYXJ5IERvbWFpbixTdWJzY3JpcHRpb24gSWQsUmVzb3VyY2UgR3JvdXAsSW5zdGFuY2UgRGF0YSxUYWdzLFVzYWdlIERhdGUsTWV0ZXIgQ2F0ZWdvcnksTWV0ZXIgU3ViY2F0ZWdvcnksTWV0ZXIgSWQsTWV0ZXIgTmFtZSxNZXRlciBSZWdpb24sQ2hhcmdlIFN0YXJ0IERhdGUsQ2hhcmdlIEVuZCBEYXRlLFF1YW50aXR5LFVuaXQsQ3VycmVuY3ksUlJQLFRvdGFsIFByZS1UYXgNCkZyb21IZXJlT24sMjQwMDkxOSwyNDAwOTE5LGZyb21oZXJlb24uY29tLGE2M2NlZGQyLWQ0MTAtNGUzMi04MmQyLWU3YzQ1NTdjNGQwYSxGSE9IYXJ2ZXN0LEZIT0hhcnZlc3Qse30sMjAxOS0wOS0wMSxMb2cgQW5hbHl0aWNzLCwwZDQzNWRmOC05Yzc1LTQ4MmUtYTg5NC1jZDY1YzIxNjBjN2EsRGF0YSBJbmdlc3Rpb24sYXVzdHJhbGlhZWFzdCwyMDE5LTA5LTAxLDIwMTktMDktMzAsMC4wMTU3Njg5MjcxMzQ1MDYwMywxIEdCLEFVRCwwLjA3MjMxMzQ2MSwwLjA2NTA4MjExNQ0KRnJvbUhlcmVPbiwyNDAwOTE5LDI0MDA5MTksZnJvbWhlcmVvbi5jb20sYTYzY2VkZDItZDQxMC00ZTMyLTgyZDItZTdjNDU1N2M0ZDBhLEZIT0hhcnZlc3QsRkhPSGFydmVzdCx7fSwyMDE5LTA5LTAyLExvZyBBbmFseXRpY3MsLDBkNDM1ZGY4LTljNzUtNDgyZS1hODk0LWNkNjVjMjE2MGM3YSxEYXRhIEluZ2VzdGlvbixhdXN0cmFsaWFlYXN0LDIwMTktMDktMDEsMjAxOS0wOS0zMCwwLjAxNjUxMDM0NzYzMjY1NDYzLDEgR0IsQVVELDAuMDc1NzEzNDgyLDAuMDY4MTQyMTM0DQpGcm9tSGVyZU9uLDI0MDA5MTksMjQwMDkxOSxmcm9taGVyZW9uLmNvbSxhNjNjZWRkMi1kNDEwLTRlMzItODJkMi1lN2M0NTU3YzRkMGEsRkhPSGFydmVzdCxGSE9IYXJ2ZXN0LHt9LDIwMTktMDktMDMsTG9nIEFuYWx5dGljcywsMGQ0MzVkZjgtOWM3NS00ODJlLWE4OTQtY2Q2NWMyMTYwYzdhLERhdGEgSW5nZXN0aW9uLGF1c3RyYWxpYWVhc3QsMjAxOS0wOS0wMSwyMDE5LTA5LTMwLDAuMDE1ODE2MTA0OTQ2NDEyNSwxIEdCLEFVRCwwLjA3MjUyOTgxMCwwLjA2NTI3NjgyOQ0KRnJvbUhlcmVPbiwyNDAwOTE5LDI0MDA5MTksZnJvbWhlcmVvbi5jb20sYTYzY2VkZDItZDQxMC00ZTMyLTgyZDItZTdjNDU1N2M0ZDBhLEZIT0hhcnZlc3QsRkhPSGFydmVzdCx7fSwyMDE5LTA5LTA0LExvZyBBbmFseXRpY3MsLDBkNDM1ZGY4LTljNzUtNDgyZS1hODk0LWNkNjVjMjE2MGM3YSxEYXRhIEluZ2VzdGlvbixhdXN0cmFsaWFlYXN0LDIwMTktMDktMDEsMjAxOS0wOS0zMCwwLjAxNjQ5MjUxMzExNDU0Njg5LDEgR0IsQVVELDAuMDc1NjMxNjk2LDAuMDY4MDY4NTI3DQpGcm9tSGVyZU9uLDI0MDA5MTksMjQwMDkxOSxmcm9taGVyZW9uLmNvbSxhNjNjZWRkMi1kNDEwLTRlMzItODJkMi1lN2M0NTU3YzRkMGEsRkhPSGFydmVzdCxGSE9IYXJ2ZXN0LHt9LDIwMTktMDktMDUsTG9nIEFuYWx5dGljcywsMGQ0MzVkZjgtOWM3NS00ODJlLWE4OTQtY2Q2NWMyMTYwYzdhLERhdGEgSW5nZXN0aW9uLGF1c3RyYWxpYWVhc3QsMjAxOS0wOS0wMSwyMDE5LTA5LTMwLDAuMDE2NTIzNjA2OTc3NTMxNTIsMSBHQixBVUQsMC4wNzU3NzQyODcsMC4wNjgxOTY4NTkNCkZyb21IZXJlT24sMjQwMDkxOSwyNDAwOTE5LGZyb21oZXJlb24uY29tLGE2M2NlZGQyLWQ0MTAtNGUzMi04M

Desired Behaviour:

Postman saves them to file which shows the CSV with Proper data likewise expecting cypress to save the CSV with valid value instead of these characters. 
Basically I am looking the way to assert the column names in the CSV file returned by the API request.

**Comment by jennifer-shehane** [comment]
There is a recipe about reading csv files here that may be helpful: https://github.com/cypress-io/cypress-example-recipes/tree/372d13039ba5ee13dc476f1bf2a264990dd65b15/examples/testing-dom__csv-table

`utf8` is a default encoding, so you shouldn't need to pass that as an argument to `cy.writeFile()`. `cy.writeFile()` essentially uses `fs` under the hood though, so I don't think this is likely the issue. 

Likely the encoding is not correct when it comes as the API response, but again, I can't say because you did not include the full code of where you are saving the content to the csv file to begin with from the API response.

**Comment by Gudguntivv** [comment]
how to write values to particular cells in CSV ??? ie.[4,4]

**Comment by NafeesathShama** [comment]
How this issue has been resolved? My API response is not in a readable format. But the downloaded response xl file from postman is in a readable format. 
My Cypress code is unable to open the downloaded file due to this.
Note: response in Postman also in unreadable format, Only downloaed responsebody-  excel file is in readable format

**Comment by nagash77** [comment]
Hi @NafeesathShama , if you are encountering an error please open a new issue with a reproducible example and we will be happy to investigate.  Also, check out our [community chat](https://discord.gg/cypress), it can be helpful for debugging or answering questions on how to use Cypress.
